File: 1.5-ArrayBasics/ArrayBasics_in_CPP/7-IntersectionArray.cpp
#include<bits/stdc++.h>
using namespace std;


vector<int> interSection(vector<int> arr1, vector<int> arr2){
    
    int n = arr1.size();
    int m = arr2.size();

    vector<int> ans;

    int i = 0, j = 0;

    while( i < n && j < m ){
        if(arr1[i] < arr2[j]){
            i++;
        }
        else if(arr1[i] == arr2[j]){
            ans.push_back(arr1[i]);
            i++;
            j++;
        }
        else{
            j++;
        }
    }

    return ans;
}

int main() {

    vector<int> arr1 = {1,2,2,2,3,4};
    vector<int> arr2 = {2,2,3,3};

    vector<int> ans = interSection(arr1, arr2);

    for(int i : ans){
        cout<<i<<" ";
    }

    return 0;
}

File: 1.5-ArrayBasics/ArrayBasics_in_Python/7-IntersectionArray.py
def interSection(arr1, arr2):

    n = len(arr1)
    m = len(arr2)

    ans = []

    i, j = 0, 0

    while( i < n and j < m):
        if( arr1[i] < arr2[j]):
            i += 1
        elif( arr1[i] == arr2[j] ):
            ans.append(arr1[i])
            i += 1
            j += 1
        else:
            j += 1

    return ans


arr1 = [1,4,5]
arr2 = [3,4,5]

ans = interSection(arr1, arr2)

print(ans)